### PR TITLE
feat: Change the kind from 'argo-project' to 'argocd'

### PR DESCRIPTION
**Pull request by chengleqi**
### What type of PR is this?
/kind design

### What this PR does / why we need it:
We use the 
```yaml
apiVersion: gitops.kubesphere.io/v1alpha1
kind: Application
spec:
  kind: argocd
  argoApp:
```
or 
```yaml
apiVersion: gitops.kubesphere.io/v1alpha1
kind: Application
spec:
  kind: fluxcd
  fluxApp:
```
to identify the kind of this application is `argocd` or `fluxcd`.
See also: https://github.com/kubesphere/ks-devops/pull/778

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
None
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: src/utils/form.templates.js [hunk 1]
```diff
@@ -695,9 +695,6 @@ const getCDTemplate = () => ({
   kind: 'Application',
   apiVersion: 'gitops.kubesphere.io/v1alpha1',
   metadata: {},
-  spec: {
-    kind: 'argo-project',
```
Comment (LinuxSuRen) [inline src/utils/form.templates.js:699]: Do we have any potential incompatible issues because of the name change?

Comment (chengleqi) [reply src/utils/form.templates.js:699]: I have tested it. It worked well.

Only https://github.com/kubesphere/console/blob/072792b15f24b4fd4d525dbb662b6a3b83c8656c/src/actions/cd.js#L81
And
https://github.com/kubesphere/console/blob/072792b15f24b4fd4d525dbb662b6a3b83c8656c/src/actions/cd.js#L109
actions use this template. I believe it won't affect other thing.

**Pull request merged**